UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -340,11 +349,9 @@ public int GetStepCount()
         /// </summary>
         public void ResetReward()
         {
-            m_Reward = 0f;
-            if (m_Done)
-            {
-                m_CumulativeReward = 0f;
-            }
+            Debug.Assert(m_LegacyRewardProvider != null, "LegacyRewardProvider is null and " +
```

Comment:
This gets called by the Agent (during `SendInfo()` and `AgentStep()`) so it probably needs to be more tolerant to null `m_LegacyRewardProvider`

Reply:
should this be part of the API for `IRewardProvider`?

Reply:
@vincentpierre

Reply:
I guess the idea of `GetIncrementalReward` is that it is reset when the function is called.  So maybe it's just a no-op for now.

Reply:
or these can be removed, and the legacy reward provider takes care of this